core/dbt/events/types.py

```diff
@@ -413,6 +418,43 @@ def message(self) -> str:
         return warning_tag(f"Deprecated functionality\n\n{description}")
 
 
+class SpacesInModelNameDeprecation(DynamicLevel):
+    def code(self) -> str:
+        return "D014"
+
+    def message(self) -> str:
+        version = ".v" + self.model_version if self.model_version else ""
+        description = (
+            f"Model `{self.model_name}{version}` has spaces in its name. This is deprecated and "
+            "may cause errors when using dbt."
+        )
+
+        if self.level == EventLevel.ERROR.value:
+            description = _error_tag(description)
+        elif self.level == EventLevel.WARN.value:
+            description = warning_tag(description)
+
+        return line_wrap_message(description)
+
+
+class TotalModelNamesWithSpacesDeprecation(DynamicLevel):
+    def code(self) -> str:
+        return "D015"
+
+    def message(self) -> str:
+        description = f"Found {self.count_invalid_names} models with spaces in their names, which is deprecated."
```

Comment:
Since this event fires when there are a non-zero number of models with spaces you need to account for there being only 1 model in how the sentence is structured.  ie "Found 1 model..." instead of "Found 1 models..."

Reply:
Restructured the whole sentence so that we could just use `model(s)` previously there were a bunch of words that would have to be interpolated depending on whether there was 1 problematic name or 2+ problematic names.